pkg/api/mlflow/service/artifact/service.go

```diff
@@ -58,3 +59,33 @@ func (s Service) ListArtifacts(
 
 	return run.ArtifactURI, artifacts, nil
 }
+
+// GetArtifact handles business logic of `GET /get-artifact` endpoint.
+func (s Service) GetArtifact(
+	ctx context.Context, req *request.GetArtifactRequest,
+) (io.ReadCloser, error) {
+	if err := ValidateGetArtifactRequest(req); err != nil {
+		return nil, err
+	}
+
+	run, err := s.runRepository.GetByID(ctx, req.GetRunID())
+	if err != nil {
+		return nil, api.NewInternalError("unable to find run '%s': %s", req.GetRunID(), err)
+	}
+	if run == nil {
+		return nil, api.NewResourceDoesNotExistError("unable to find run '%s'", req.GetRunID())
+	}
+	artifactStorage, err := s.artifactStorageFactory.GetStorage(run.ArtifactURI)
+	if err != nil {
+		return nil, api.NewInternalError("run with id '%s' has unsupported artifact storage", run.ID)
+	}
+
+	artifactReader, err := artifactStorage.GetArtifact(
+		run.ArtifactURI, req.Path,
+	)
+	if err != nil {
+		return nil, api.NewInternalError("error getting artifact URI from storage")
```

Comment:
error message a bit misleading. I think it could be something like -> `error getting artifact object for URI: %s from storage`

Reply:
done